### PR TITLE
Fix(config): better resolving of components on windows

### PR DESCRIPTION
**Pull request by elevatebart**

### CODE REVIEW

**Comment by sapegin** [comment]
> nocase Perform a case-insensitive match. Note: on case-insensitive filesystems, non-magic patterns will match by default, since stat and readdir will not raise errors.

If I understand this correctly, it's case-insensitive on Windows in any case, so your fix is likely relying on some filtering they do when this flag is enabled. Probably we should report it to glob as well.

File: loaders/utils/getComponentFiles.js [hunk 1]
```diff
@@ -8,7 +8,11 @@ const getFilesMatchingGlobs = (components, rootDir, ignore) =>
 		.map(listItem => {
 			// Check if the string looks like a glob pattern by using hasMagic
 			if (glob.hasMagic(listItem)) {
-				return glob.sync(path.resolve(rootDir, listItem), { ignore });
+				return glob.sync(listItem, {
+					cwd: rootDir,
+					ignore,
+					nocase: process.platform === 'win32',
```
Comment (sapegin) [inline loaders/utils/getComponentFiles.js]: Please add a comment with the explanation here ;-)

File: loaders/utils/getComponentFiles.js [hunk 2]
```diff
@@ -28,21 +32,25 @@ module.exports = function getComponentFiles(components, rootDir, ignore) {
...
 
-	let componentFiles;
-	if (isFunction(components)) {
-		componentFiles = getFilesMatchingGlobs(components(), rootDir, ignore);
-	} else if (Array.isArray(components)) {
-		componentFiles = getFilesMatchingGlobs(components, rootDir, ignore);
-	} else if (isString(components)) {
-		componentFiles = glob.sync(path.resolve(rootDir, components), { ignore });
-	} else {
+	const componentGlobsArray = isFunction(components)
```
Comment (sapegin) [inline loaders/utils/getComponentFiles.js]: Just `componentGlobs` will be enough.

File: loaders/utils/getComponentFiles.js [hunk 3]
```diff
@@ -28,21 +32,25 @@ module.exports = function getComponentFiles(components, rootDir, ignore) {
...
+		? components()
```
Comment (sapegin) [inline loaders/utils/getComponentFiles.js]: A function will be more readable than a nested ternary:

```js
const getComponentGlobs = components => {
  if (isFunction(components)) {
    return components();
  }
  // ...
}
```

File: loaders/utils/getComponentFiles.js [hunk 4]
```diff
@@ -28,21 +32,25 @@ module.exports = function getComponentFiles(components, rootDir, ignore) {
...
+		: Array.isArray(components)
+			? components
+			: isString(components)
+				? [components]
+				: 0;
```
Comment (sapegin) [inline loaders/utils/getComponentFiles.js]: `false` or `undefined` would be a more clear way to say that the input was incorrect.

File: loaders/utils/getComponentFiles.js [hunk 5]
```diff
@@ -28,21 +35,26 @@ module.exports = function getComponentFiles(components, rootDir, ignore) {
...
 
-	let componentFiles;
-	if (isFunction(components)) {
-		componentFiles = getFilesMatchingGlobs(components(), rootDir, ignore);
-	} else if (Array.isArray(components)) {
-		componentFiles = getFilesMatchingGlobs(components, rootDir, ignore);
-	} else if (isString(components)) {
-		componentFiles = glob.sync(path.resolve(rootDir, components), { ignore });
-	} else {
+	const getComponentGlobs = components => {
```
Comment (sapegin) [inline loaders/utils/getComponentFiles.js]: Could you please move this function to the top of the  file?

**Comment by elevatebart** [comment]
Done ;)

**Pull request merged**